tests/app/main/test_frameworks.py

```diff
@@ -2316,3 +2316,198 @@ def test_framework_dashboard_shows_returned_agreement_details(
             assert '<tdclass="summary-item-field"><span><p>signername</p><p>signerrole</p></span></td>' in page_without_whitespace  # noqa
             assert '<tdclass="summary-item-field"><span><p>User</p><p>email@email.com</p><p>Sunday10July2016at22:20</p></span></td>' in page_without_whitespace  # noqa
             assert '<tdclass="summary-item-field-first"><span>WaitingforCCStocountersign</span></td>' in page_without_whitespace  # noqa
+
+
+@mock.patch('app.main.views.frameworks.data_api_client', autospec=True)
+class TestContractVariation(BaseApplicationTest):
```

Comment:
There isn't.  The app will error and 500 in that case, which I think is OK because the API call should never be made if the variation is already agreed - it would be a genuine application error if it did.